src/cplscheme/tests/SerialImplicitCouplingSchemeTest.cpp

```diff
@@ -450,69 +450,64 @@ BOOST_AUTO_TEST_CASE(testExtrapolateData)
 
   scheme.addDataToSend(data, mesh, true);
   scheme.setExtrapolationOrder(1);
-  scheme.setupDataMatrices(scheme.getSendData());
+  scheme.setupDataMatrices();
   CouplingData *cplData = scheme.getSendData(dataID);
   BOOST_CHECK(cplData); // no nullptr
   BOOST_TEST(cplData->values().size() == 1);
-  BOOST_TEST(cplData->oldValues.cols() == 2);
-  BOOST_TEST(cplData->oldValues.rows() == 1);
+  BOOST_TEST(cplData->lastIteration().size() == 1);
   BOOST_TEST(testing::equals(cplData->values()(0), 0.0));
-  BOOST_TEST(testing::equals(cplData->oldValues(0, 0), 0.0));
-  BOOST_TEST(testing::equals(cplData->oldValues(0, 1), 0.0));
```

Comment:
We now test the `oldValues` in the corresponding unit test for the waveform and just look at the result of the extrapolation that the cplscheme has to care about.